### PR TITLE
54 admin applications section

### PR DESCRIPTION
**Pull request by AhmedAbuSharar**

### CODE REVIEW

File: client/.eslintrc.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
         "react-hooks/rules-of-hooks": "error",
         "react-hooks/exhaustive-deps": "warn",
         "import/prefer-default-export": "off",
+        "camelcase": "off",
```
Comment (Raghad-Mezied) [inline client/.eslintrc.json]: remove this role

File: client/src/components/admin/ApplicationCard.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  ownerId: number;
+  licenseNumber: number;
+  pharmacyName: string;
+};
+interface Props {
+  card: Card;
+  setApproved: () => Promise<void>;
+  setRejected: () => Promise<void>;
+}
+
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationCard.tsx]: Move the types

File: client/src/components/admin/ApplicationCard.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import CloseIcon from '@mui/icons-material/Close';
+import { Box, Typography } from '@mui/material';
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationCard.tsx]: Import them once, as you did in line 4

File: client/src/components/admin/ApplicationSection.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  image: string;
+  description: string;
+  license_number: number;
+  status: string;
+  owner_id: number;
+  owner_name: string;
+  owner_img: string;
+  createdAt: string;
+  updatedAt: string;
+};
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationSection.tsx]: Update the import and move the type

File: client/src/components/admin/ApplicationSection.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+const ApplicationSection = () => {
+  const [cards, setCards] = useState<row[]>([] as row[]);
+  const getData = async () => {
+    const allApplications = await axios.post(
+      '/admin/pharmacies?status=Pending',
+    );
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationSection.tsx]: Why post? you are getting the data

File: client/src/components/admin/ApplicationSection.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    setCards(allApplications.data.rows);
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationSection.tsx]: Destructure the data

File: client/src/components/admin/ApplicationSection.tsx [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  };
+  const setStatus = async (status: string, id: number) => {
+    const allApplications = await axios.post(`/admin/pharmacy/${id}`, { status });
+    setCards(allApplications.data.rows);
+  };
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationSection.tsx]: You are updating the status here so its put not post, right?

File: client/src/components/admin/ApplicationSection.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+      <Typography
+        sx={{ margin: '4rem 10%', fontSize: '3rem' }}
+        variant="h1"
+        gutterBottom
+      >
+        Applications
+      </Typography>
+      {getAllTasksApplications(cards)}
+    </Box>
+  );
```
Comment (Raghad-Mezied) [inline client/src/components/admin/ApplicationSection.tsx]: Add pagination and loading

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Comment by Raghad-Mezied** [comment]
Don't forget to add a screenshot from your work in the description of the pull request.

**Comment by AhmedAbuSharar** [comment]
image

**Review by Nader-SH — DISMISSED** [review]

**Review by Ibtisam-Hemmo — DISMISSED** [review]

**Pull request merged**